data/templates/secure_erase.py

```diff
@@ -0,0 +1,341 @@
+    :param cmd: a list includes secure erase command argument
+    :return: a dict includes SE command exitcode and SE message
+    '''
+    log_file = disk_name.split("/")[-1] + ".log"  #log file for sdx will be sdx.log
+    log = open(log_file, "a")
+    record_timestamp(log=log, action="start")
+    disk_size = get_disk_size(disk_name, log)
+    back_skip = str(int(disk_size)/512-int(FRONT_SKIP))
+    mark_on_disk(disk_name, log, False, back_skip)  #Create marks on disk
+    exit_status = robust_check_call(cmd=cmd, log=log)
+    mark_on_disk(disk_name, log, True, back_skip)  #Verify marks on disk
+    record_timestamp(log=log, action="complete")
+    log.close()
+    return exit_status
+
+def sg_format_secure_erase(disk_name):
+    '''
+    Secure erase using sg_format tool
+    :param disk_name: an list includes all disk_list will be erased
+    :return: a dict includes SE command exitcode and SE message
+    '''
+    glist_erase_bit = ARG_LIST.o
+    command = ["sg_format", "-v", "--format",
+               "--cmplst=" + glist_erase_bit, disk_name]
+    return secure_erase_base(disk_name, cmd=command)
+
+
+def sg_sanitize_secure_erase(disk_name):
+    '''
+    Secure erase using sg_sanitize tool
+    :param disk_name: an list includes all disk_list will be erased
+    :return: a dict includes SE command exitcode and SE message
+    '''
+    option = ARG_LIST.o
+    command = ["sg_sanitize", "-v", option, disk_name]
+    return secure_erase_base(disk_name, cmd=command)
+
+def scrub_secure_erase(disk_name):
+    '''
+    Secure erase using scrub tool
+    :param disk_name: an list includes all disk_list will be erased
+    :return: a dict includes SE command exitcode and SE message
+    '''
+    option = ARG_LIST.o
+    command = ["scrub", "-f", "-p", option, disk_name] # -f is to force erase
+    return secure_erase_base(disk_name, cmd=command)
+
+
+if __name__ == '__main__':
+    TOOL_MAPPER = {
+        "scrub": scrub_secure_erase,
+        "hdparm" : hdparm_secure_erase,
+        "sg_format": sg_format_secure_erase,
+        "sg_sanitize": sg_sanitize_secure_erase
+    }
+    tool = ARG_LIST.t
+    disk_list = ARG_LIST.d.split(",")
+    assert tool in ["scrub", "hdparm", "sg_format", "sg_sanitize"], \
+        "Secure erase tool is not supported"
+    assert disk_list != "none", "No disk is specified for secure erase"
+
+    #Get process count we should started
+    user_count = len(disk_list)
+    cpu_thread_count = cpu_count()
+    if user_count > cpu_thread_count:
+        process_count = cpu_thread_count
+    else:
+        process_count = user_count
+    pool = Pool(process_count)
+
+    erase_output_list = []
+    for disk in disk_list:
+        #Create log files for status check
+        erase_output = {"seMethod": tool, "disk": disk}
+        result = pool.apply_async(TOOL_MAPPER[tool], args=(disk,))
+        erase_output["poolExitStatus"] = result
+        erase_output_list.append(erase_output)
```

Comment:
Since here is a write, any lock need for the share resource `erase_output_list`?